schemas/cs3/tbl/magic_bo.ksy

```diff
@@ -0,0 +1,53 @@
+meta:
+  id: magic_bo
+  endian: le
+doc: |
+  Effects and parameter of Brave Order count effects.
+
+  Present in t_magic.tbl.
+seq:
+  - id: skill_id
+    type: u2
+  - id: bo_effect_id_1
+    type: u2
+  - id: bo_effect_param_1_1
+    type: s4
+  - id: bo_effect_param_2_1
+    type: s4
+  - id: bo_effect_param_3_1
+    type: s4
+  - id: bo_effect_id_2
+    type: u2
+  - id: bo_effect_param_1_2
+    type: s4
+  - id: bo_effect_param_2_2
+    type: s4
+  - id: bo_effect_param_3_2
+    type: s4
+  - id: bo_effect_id_3
+    type: u2
+  - id: bo_effect_param_1_3
+    type: s4
+  - id: bo_effect_param_2_3
+    type: s4
+  - id: bo_effect_param_3_3
+    type: s4
+  - id: bo_effect_id_4
+    type: u2
+  - id: bo_effect_param_1_4
+    type: s4
+  - id: bo_effect_param_2_4
+    type: s4
+  - id: bo_effect_param_3_4
+    type: s4
+  - id: bo_effect_id_4
+    type: u2
+  - id: bo_effect_param_1_4
+    type: s4
+  - id: bo_effect_param_2_4
+    type: s4
+  - id: bo_effect_param_3_4
+    type: s4
+  - id: count_amount
+    type: u1

```

Comment:
Also, do we know what is being counted here? If yes, it should probably be documented, if not it should be renamed to `unknown_byte` (or short depending on the above comment)

Reply:
> ngth is 72.

Isn't the last 00 value just a null terminate of some kind?
If not then it's u2 yeah.

Reply:
> Also, do we know what is being counted here? If yes, it should probably be documented, if not it should be renamed to `unknown_byte` (or short depending on the above comment)

Counts are how brave order turns are referred to. Ebon Crest lasts for 4 counts, Raging Fire lasts for 6 counts etc.

Reply:
> Isn't the last 00 value just a null terminate of some kind?

The value does seem to be unused, so I guess we can make this field a `u1` and add a `contents: [0x00]` field after that.

> Counts are how brave order turns are referred to.

I'd suggest `turns` or `duration` instead.

Reply:
> I'd suggest turns or duration instead
The term count has been in use for years now, there's little to no reason to change it.
And calling it turns or duration is confusing in it of itself since the duration of buffs are calculated in turns which are different from counts which are effectively global buff turns instead.

Reply:
Is that an in-game term? If it is we should use it of course, but otherwise I don't think it's very clear tbh. Should also be noted that the game already overloads "turn" (off the cuff it means something different for buffs than for gladiator belt CP regen). I'm generally in favour of using in-game terms but it might also make sense to choose more precise terms.